python/pyspark/ml/util.py

```diff
@@ -536,10 +536,8 @@ def __get_class(clazz: str) -> Type[RL]:
         """
         parts = clazz.split(".")
         module = ".".join(parts[:-1])
-        m = __import__(module)
-        for comp in parts[1:]:
-            m = getattr(m, comp)
-        return m
+        m = __import__(module, fromlist=[parts[-1]])
+        return getattr(m, parts[-1])
```

Comment:
Hm, I think this changes the semantic. Previously, it returned the left-most import but not it returns right-most package. For example,

```python
>>> __import__("pyspark", fromlist=['sql'])
<module 'pyspark' from '/.../spark/python/pyspark/__init__.py'>
>>> m = __import__("pyspark")
>>> getattr(m, "sql")
<module 'pyspark.sql' from '/.../spark/python/pyspark/sql/__init__.py'>
```

Reply:
```python
clazz = "pyspark"

parts = clazz.split(".")
module = ".".join(parts[:-1])
m = __import__(module)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Empty module name

parts = clazz.split(".")
module = ".".join(parts[:-1])
m = __import__(module, fromlist=[parts[-1]])

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Empty module name
```

```python
clazz = "pyspark.sql"

# previous
parts = clazz.split(".")
module = ".".join(parts[:-1])
m = __import__(module)
for comp in parts[1:]:
    m = getattr(m, comp)

m
<module 'pyspark.sql' from '/Users/yikun/spark/python/pyspark/sql/__init__.py'>

# new
parts = clazz.split(".")
module = ".".join(parts[:-1])
m = __import__(module, fromlist=[parts[-1]])
getattr(m, parts[-1])
<module 'pyspark.sql' from '/Users/yikun/spark/python/pyspark/sql/__init__.py'>
```

BTW, the `from_list` value seems useless in our case. See also: https://stackoverflow.com/questions/2724260/why-does-pythons-import-require-fromlist

Reply:
The original error from here is about type different error. As the previous function might return a python module type `m = __import__(module)`, then finally `return m` which can not be called, that's due to the [following line ](https://github.com/apache/spark/blob/master/python/pyspark/ml/util.py#L546). And the defined return-value is a Type[RL], they are different. So I think this func should return a python Class at least. That's why I change it.